app_feup/lib/view/Widgets/bus_stop_search.dart

```diff
@@ -89,23 +89,25 @@ class BusStopSearch extends SearchDelegate<String> {
           width: 100.0,
         ),
         actions: [
-          FlatButton(
+          TextButton(
               child: Text('Cancelar',
                   style: Theme.of(context)
                       .textTheme
-                      .display1
+                      .headline4
                       .apply(color: Theme.of(context).primaryColor)),
               onPressed: () => Navigator.pop(context)),
-          FlatButton(
+          TextButton(
               child: Text('Confirmar',
                   style: Theme.of(context)
                       .textTheme
-                      .display1
+                      .headline4
                       .apply(color: Theme.of(context).accentColor)),
-              color: Theme.of(context).primaryColor,
-              shape: RoundedRectangleBorder(
-                  borderRadius:  BorderRadius.circular(10.0),
-                  side: BorderSide(color: Theme.of(context).primaryColor)),
+              style: TextButton.styleFrom(
+                primary: Theme.of(context).backgroundColor,
```

Comment:
Before I got locked out of the app due to a login bug, I was able to try this button and it was still the same color, that is because you switched _primary_ with _backgroundColor_ in the _Theme_ call, not on the property. It should be: 
`backgroundColor: Theme.of(context).primary,`

Reply:
So the line you told me to change is wrong and should be changed back to primary?

Reply:
No, what you had initially was: `primary: Theme.of(context).primaryColor,` 
It was then changed to `primary: Theme.of(context).backgroundColor`, and it should be: `backgroundColor: Theme.of(context).primaryColor,` 
I now realize I reffered to _primary_ as a property instead of an argument to styleFrom() and that was confusing, I'm sorry.